sway-lib-std/src/storage.sw

```diff
@@ -83,15 +114,59 @@ pub fn get<T>(key: b256) -> T {
     }
 }
 
+/// A persistent key-value pair mapping struct
 pub struct StorageMap<K, V> {}
 
 impl<K, V> StorageMap<K, V> {
+    /// Inserts a key-value pair into the map.
+    /// 
+    /// # Arguments
+    ///
+    /// * `key` - The key to which the value is paired
+    /// * `value` - The value to be stored
+    ///
+    /// # Examples
+    /// 
+    /// ```
+    /// storage {
+    ///     map: StorageMap = StorageMap {} 
+    /// }
+    /// 
+    /// fn foo() {
+    ///     let key = 5_u64;
+    ///     let value = true;
+    ///     storage.map.insert(key, value);
+    ///     let retrieved_value = storage.map.get(key);
+    ///     assert(value == retrieved_value);
+    /// }
+    /// ```
     #[storage(write)]
     fn insert(self, key: K, value: V) {
         let key = sha256((key, __get_storage_key()));
         store::<V>(key, value);
     }
 
+    /// Retrieves a value previously stored using a key
```

Comment:
We should probably describe the behavior when `key` has not been paired yet to anything in which case we get back "zeros". I've described this previously in https://fuellabs.github.io/sway/v0.30.1/common-collections/storage_map.html#accessing-values-in-a-storage-map

This will hopefully improve in the future.